app.rb

```diff
@@ -0,0 +1,70 @@
+require 'sinatra'
+require "./lib/peep"
+require "./lib/user"
+
+
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+
+  attr_reader :username, :new_username, :password
+
+  get '/' do
+      erb(:sign_in)
+  end
+
+
+  post '/log_in' do
+    session[:username] = params[:username]
+    session[:password] = params[:password]
+    if User.find_user(username: session[:username]) == false
+      redirect('/no_user')
+    else
+      redirect('/check_password')
+    end
+  end
+
+  get '/no_user' do
+    erb(:no_user_sign_up)
+  end
+
+  get '/check_password' do
```

Comment:
might have been better to have this functionality in the model to keep the controllers skinny, however, it does provide extra error messages.